streamer/src/nonblocking/quic.rs

```diff
@@ -99,8 +99,11 @@ pub fn spawn_server(
     coalesce: Duration,
 ) -> Result<(Endpoint, Arc<StreamStats>, JoinHandle<()>), QuicServerError> {
     info!("Start {name} quic server on {sock:?}");
-    let (config, _cert) = configure_server(keypair, gossip_host)?;
-
+    let (config, _cert) = configure_server(
+        keypair,
+        gossip_host,
+        max_staked_connections.saturating_add(max_unstaked_connections) as u32,
```

Comment:
Given,
> This is also effective against connections still in "Connecting" mode.

Wouldn't this block a high staked node from evicting any of lower staked nodes if we have already hit this limit?

Reply:
Yes, it would. This fix main goal was to avoid building excessive in-progress connections as stop-gap measure. Finer control of in-progress connections will be addressed in follow-on PRs.

Reply:
If a higher staked node is blocked by lower staked nodes then we shouldn't merge this.

At the very least this limit should be bigger than sum of `max_{staked,unstaked}_connections. Maybe twice as big.

Reply:
I am testing the effectiveness of the parameter will close this PR for now.

Reply:
Have we confirmed that this limits connections early in the connection process? Also, perhaps it might be worth investigating imposing limits per-IP?